DQMServices/Core/src/DQMStore.cc

```diff
@@ -99,15 +99,30 @@ namespace dqm::implementation {
       MonitorElementData medata;
       medata.key_.path_ = path;
       medata.key_.kind_ = kind;
-      medata.key_.scope_ = this->scope_;
+
+      bool pathInList = false;
+      if (not store_->MEsToSave_.empty()) {
```

Comment:
```suggestion
      if (not store_->MEsToSave_.empty()) {
         bool pathInList = false;
```
(define the variable in the smallest possible scope)